pkg/metrics/metrics.go

```diff
@@ -0,0 +1,181 @@
+// Package metrics provides functionality for recording and serving
+// metric information related to AWS API calls and internal driver operations.
+package metrics
+
+import (
+	"net/http"
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// Metric types
+	AWS metricType = iota
+	Internal
+)
+const (
+	// AWS API metrics
+	awsDurationMetric = "cloudprovider_aws_api_request_duration_seconds"
+	awsThrottleMetric = "cloudprovider_aws_api_throttled_requests_total"
+	awsErrorMetric    = "cloudprovider_aws_api_request_errors"
+
+	// Internal driver metrics
+	internalDurationMetric = "ebs_csi_aws_com_duration_seconds"
+	internalValueMetric    = "ebs_csi_aws_com_operation"
+	internalErrorMetric    = "ebs_csi_aws_com_errors_total"
+)
+
+var (
+	// r is the singleton instance of metricRecorder.
+	r *metricRecorder
+
+	initialize sync.Once
+)
+
+type metricType int
+
+type metricRecorder struct {
+	registry metrics.KubeRegistry
+	metrics  map[metricType]metricFields
+}
+
+type metricFields struct {
+	duration *metrics.HistogramVec
+	value    *metrics.HistogramVec
```

Comment:
Probably related to the `internalValue` above, but is this meant to be a count of all occurrences? Should this value be called `count`?